src/plugins/data/public/query/query_string/language_service/lib/query_result.tsx

```diff
@@ -28,21 +28,66 @@ export interface QueryStatus {
     statusCode?: number;
   };
   elapsedMs?: number;
+  startTime?: number;
 }
 
+// This is the time in milliseconds that the query will wait before showing the loading spinner
```

Comment:
Why would it wait for this huge amount of time tow show the spinner if there really is something loading?

Reply:
@ashwin-pc requests that we wait for some time before we show the loading indicator.

Reply:
Yeah for most usecases the query runs in milliseconds, its only when the query takes longer than a few seconds that you are left wondering whats happening. I think 3 seconds is a good buffer to wait before indicating the user that hey your query is running, its just taking a bit longer than expected.